src/main.cpp

```diff
@@ -2940,7 +2940,7 @@ bool static DisconnectTip(CValidationState &state, bool fBare = false) {
     int64_t nStart = GetTimeMicros();
     {
         CCoinsViewCache view(pcoinsTip);
-        if (DisconnectBlock(block, state, pindexDelete, view, fAddressIndex) != DISCONNECT_OK)
+        if (DisconnectBlock(block, state, pindexDelete, view, true) != DISCONNECT_OK)
```

Comment:
Why did this change to always being true, whereas before it might have been false?

Reply:
See my comment below where I explain this; it's really just so that `fAddressIndex` can appear in the code right at the place where it has an effect (to make it easier for readers).